modules/structs/repo.go

```diff
@@ -70,38 +78,36 @@ type CreateRepoOption struct {
 // EditRepoOption options when editing a repository's properties
 // swagger:model
 type EditRepoOption struct {
-	// Name of the repository
-	//
-	// required: true
+	// name of the repository
 	// unique: true
-	Name *string `json:"name" binding:"Required;AlphaDashDot;MaxSize(100)"`
-	// A short description of the repository.
+	Name *string `json:"name,omitempty" binding:"OmitEmpty;AlphaDashDot;MaxSize(100);"`
+	// a short description of the repository.
 	Description *string `json:"description,omitempty" binding:"MaxSize(255)"`
-	// A URL with more information about the repository.
+	// a URL with more information about the repository.
 	Website *string `json:"website,omitempty" binding:"MaxSize(255)"`
-	// Either `true` to make the repository private or `false` to make it public.
-	// Note: You will get a 422 error if the organization restricts changing repository visibility to organization
+	// either `true` to make the repository private or `false` to make it public.
+	// Note: you will get a 422 error if the organization restricts changing repository visibility to organization
 	// owners and a non-owner tries to change the value of private.
 	Private *bool `json:"private,omitempty"`
-	// Either `true` to enable issues for this repository or `false` to disable them.
-	EnableIssues *bool `json:"enable_issues,omitempty"`
-	// Either `true` to enable the wiki for this repository or `false` to disable it.
-	EnableWiki *bool `json:"enable_wiki,omitempty"`
-	// Updates the default branch for this repository.
+	// either `true` to enable issues for this repository or `false` to disable them.
+	HasIssues *bool `json:"has_issues,omitempty"`
+	// either `true` to enable the wiki for this repository or `false` to disable it.
+	HasWiki *bool `json:"has_wiki,omitempty"`
+	// sets the default branch for this repository.
 	DefaultBranch *string `json:"default_branch,omitempty"`
-	// Either `true` to allow pull requests, or `false` to prevent pull request.
-	EnablePullRequests *bool `json:"enable_pull_requests,omitempty"`
-	// Either `true` to ignore whitepace for conflicts, or `false` to not ignore whitespace. `enabled_pull_requests` must be `true`.
-	IgnoreWhitespaceConflicts *bool `json:"ignore_whitespace,omitempty"`
-	// Either `true` to allow merging pull requests with a merge commit, or `false` to prevent merging pull requests with merge commits. `enabled_pull_requests` must be `true`.
+	// either `true` to allow pull requests, or `false` to prevent pull request.
+	AllowPullRequests *bool `json:"allow_pull_requests,omitempty"`
```

Comment:
Why not follow the above with the name `HasPullRequest`?

Reply:
To me this sounds kind of odd, has having a wiki or having issues is different then allowing pull requests, and all the other pull request options are also "Allow"

Reply:
Imho allow is also better

Reply:
Just now saw @lafriks comment. In the maintainers group on Discord others said has_ was better. I went with has_. See commit 3f44c1c419deebd343e8af9810413f7a3943dec0